From ad7e3386a512bc43cbf285f3ee8eb3fca6504808 Mon Sep 17 00:00:00 2001
From: Joseph C Wang <joequant@gmail.com>
Date: Sat, 4 Apr 2026 18:17:55 +0800
Subject: [PATCH] use badname if @ is in username

---
 tljh/user.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tljh/user.py b/tljh/user.py
index b03903f1..43af3880 100644
--- a/tljh/user.py
+++ b/tljh/user.py
@@ -26,7 +26,13 @@ def ensure_user(username):
         # User doesn't exist, time to create!
         pass
 
-    subprocess.check_call(["useradd", "--create-home", username])
+    try:
+        subprocess.check_call(["useradd", "--create-home", username])
+    except subprocess.CalledProcessError:
+        if "@" in username:
+            subprocess.check_call(["useradd", "--create-home", "--badname", username])
+        else:
+            raise
 
     subprocess.check_call(["chmod", "o-rwx", expanduser(f"~{username}")])
 
